tests/sentry/integrations/api/endpoints/test_data_forwarding_details.py

```diff
@@ -0,0 +1,474 @@
+from sentry.integrations.models.data_forwarder import DataForwarder
+from sentry.integrations.models.data_forwarder_project import DataForwarderProject
+from sentry.integrations.types import DataForwarderProviderSlug
+from sentry.testutils.cases import APITestCase
+from sentry.testutils.silo import region_silo_test
+
+
+@region_silo_test
+class DataForwardingDetailsEndpointTest(APITestCase):
+    endpoint = "sentry-api-0-organization-forwarding-details"
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.login_as(user=self.user)
+
+
+@region_silo_test
+class DataForwardingDetailsPutTest(DataForwardingDetailsEndpointTest):
+    method = "PUT"
+
+    def test_update_data_forwarder(self) -> None:
+        data_forwarder = self.create_data_forwarder(
+            provider=DataForwarderProviderSlug.SEGMENT,
+            config={"write_key": "old_key"},
+            is_enabled=True,
+        )
+
+        payload = {
+            "provider": DataForwarderProviderSlug.SEGMENT,
+            "config": {"write_key": "new_key"},
+            "is_enabled": False,
+            "enroll_new_projects": True,
+            "project_ids": [self.project.id],
+        }
+
+        response = self.get_success_response(
+            self.organization.slug, data_forwarder.id, status_code=200, **payload
+        )
+
+        assert response.data["config"] == {"write_key": "new_key"}
```

Comment:
I would also add a check that the previous old key, and is_enabled were set properly before the API call, that way we know they changed 👍